src/platform/posix/tcp_ip_channel.c

```diff
@@ -261,56 +296,65 @@ static lf_ret_t TcpIpChannel_open_connection(NetworkChannel *untyped_self) {
 static lf_ret_t TcpIpChannel_send_blocking(NetworkChannel *untyped_self, const FederateMessage *message) {
   LF_DEBUG(NET, "TcpIpChannel: Send blocking");
   TcpIpChannel *self = (TcpIpChannel *)untyped_self;
+  lf_ret_t lf_ret = LF_ERR;
 
-  int socket;
-
-  // based if this super is in the server or client role we need to select different sockets
-  if (self->server) {
-    socket = self->client;
-  } else {
-    socket = self->fd;
-  }
-
-  // serializing protobuf into buffer
-  int message_size = serialize_to_protobuf(message, self->write_buffer, TCP_IP_CHANNEL_BUFFERSIZE);
-
-  if (message_size < 0) {
-    LF_ERR(NET, "TcpIpChannel: Could not encode protobuf");
-    return LF_ERR;
-  }
+  pthread_mutex_lock(&self->state_mutex);
```

Comment:
If the worker thread is constantly polling the socket, then I think he will detect the closed socket very quickly anyways and we would need to handle this from the `send_blocking` side and we wouldn't need to lock any mutex here.

However, I am unsure whether we want to have the worker thread busy-poll the receive socket. Is it possible for the worker thread to both block on the receiving socket AND a condition variable? And then we could signal that condition variable from `send_blocking` if we fail to send. That would wake up the worker thread, and then let it handle the updating of state etc?